tests/test_qsim_device.py

```diff
@@ -380,34 +378,24 @@ def test_expval_single_wire_with_parameters(
         assert np.isclose(res, expected_output, **tol)
 
     @pytest.mark.parametrize(
-        "observable1,observable2",
+        "observable1,observable2,expected_output",
         [
-            (qml.Identity, qml.PauliZ),
+            (qml.Identity, qml.PauliZ, -1),
```

Comment:
> It would be nice to test a single qml.Identity too, as that is a separate case in the condition used in the expval method.

Still a good idea to cover the `observable.name == "Identity"` case.